llvm/lib/Target/NVPTX/NVPTXUtilities.cpp

```diff
@@ -296,37 +298,44 @@ bool isKernelFunction(const Function &F) {
   return (x == 1);
 }
 
-bool getAlign(const Function &F, unsigned index, unsigned &align) {
+MaybeAlign getAlign(const Function &F, unsigned Index) {
+  // First check the alignstack metadata
+  if (MaybeAlign AlignStack =
```

Comment:
Nit: In some places you're using StackAligh, in others AlignStack. It might be good to settle on one variant when we're dealing with the same info.